src/modules/commander/commander.cpp

```diff
@@ -430,6 +433,13 @@ transition_result_t arm_disarm(bool arm, const int mavlink_fd_local, const char
 {
 	transition_result_t arming_res = TRANSITION_NOT_CHANGED;
 
+	// For HIL platforms, require that simulated sensors are connected
+	if (autostart_id >= HIL_ID_MIN && autostart_id <= HIL_ID_MAX
+			&& status.hil_state != vehicle_status_s::HIL_STATE_ON) {
+		mavlink_and_console_log_critical(mavlink_fd_local, "HIL platform: Connect to simulator before arming");
```

Comment:
It looks like the function should also check if the user is trying to go from disarm -> arm